tehuti.py

```diff
@@ -168,6 +170,28 @@ def run(self):
         return [sum(vals) / self.number for vals in self.usage_log]
 
 
+class AccuracyMetric(Metric):
+    class Context(object):
+        pass
+
+    def __init__(self, body, setup, name=None):
+        self.body = body
+        self.setup = setup
+        self.name = name
+        self.log = []
+
+    def id(self):
+        return 'accuracy-{}'.format(self.name or self.body.func_name)
+
+    def run(self):
+        context = AccuracyMetric.Context()
+        setup = lambda: self.setup(context)
+        func = lambda: self.body(context)
```

Comment:
This is much simpler and easier to digest now - thank you.

Can you further simplify by getting rid of the lambdas and just calling `self.setup` and `self.body` directly?